addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -491,7 +491,11 @@ const UserValueWidget = Widget.extend({
                 this._triggerWidgetsValues.push(...dataValue.split(/\s*,\s*/g));
             } else if (validMethodNames.includes(key)) {
                 this._methodsNames.push(key);
-                this._methodsParams.optionsPossibleValues[key] = dataValue.split(/\s*\|\s*/g);
+                if (key === "setVisibilityDependency") {
+                    this._methodsParams.optionsPossibleValues[key] = [dataValue];
+                } else {
```

Comment:
It looks like the right fix, but this is an unusual place for that kind of fix, so this would require a comment in the code I think. :thinking: 

And apart from that, this one definitely needs a test. (Add some steps, or maybe just rewrite some existing steps in `website/static/tests/tours/website_form_editor.js`.)

Reply:
Or actually, maybe this is not the right fix. Maybe you should find a way to encode the value with an HTML entity for '|' on the form side.